lib/ansible/modules/cloud/huawei/hwc_network_vpc.py

```diff
@@ -0,0 +1,520 @@
+    not_format_keys = re.findall("={marker}", link)
+    none_values = re.findall("=None", link)
+
+    if not (not_format_keys or none_values):
+        r = fetch_resource(session, link)
+        if r is None:
+            return ""
+        r = r.get('vpcs', [])
+        ids = [
+            i.get('id') for i in r if i.get('name', '') == name
+        ]
+        if not ids:
+            return ""
+        elif len(ids) == 1:
+            return ids[0]
+        else:
+            module.fail_json(msg="Multiple resources with same name are found")
+    elif none_values:
+        module.fail_json(
+            msg="Can not find id by name because url includes None")
+    else:
+        p = {'marker': ''}
+        ids = set()
+        while True:
+            r = fetch_resource(session, link.format(**p))
+            if r is None:
+                break
+            r = r.get('vpcs', [])
+            if r == []:
+                break
+            for i in r:
+                if i.get('name') == name:
+                    ids.add(i.get('id'))
+            if len(ids) >= 2:
+                module.fail_json(
+                    msg="Multiple resources with same name are found")
+
+            p['marker'] = r[-1].get('id')
+
+        return ids.pop() if ids else ""
+
+
+@link_wrapper
+def list_link(session, extra_data=None):
+    url = "{endpoint}vpcs?limit={limit}&marker={marker}"
+
+    combined = session.module.params.copy()
+    if extra_data:
+        combined.update(extra_data)
+
+    combined['endpoint'] = session.get_service_endpoint('vpc')
+
+    return url.format(**combined)
+
+
+@link_wrapper
+def self_link(session):
+    url = "{endpoint}vpcs/{id}"
+
+    combined = session.module.params.copy()
+    combined['endpoint'] = session.get_service_endpoint('vpc')
+
+    return url.format(**combined)
+
+
+@link_wrapper
+def collection(session):
+    url = "{endpoint}vpcs"
+
+    combined = session.module.params.copy()
+    combined['endpoint'] = session.get_service_endpoint('vpc')
+
+    return url.format(**combined)
+
+
+def return_if_object(module, response, success_codes, has_content=True):
+    code = response.status_code
+
+    # If not found, return nothing.
+    if code == 404:
+        return None
+
+    success_codes = [200, 201, 202, 203, 204, 205, 206, 207, 208, 226]
+    # If no content, return nothing.
+    if code in success_codes and not has_content:
+        return None
+
+    result = None
+    try:
+        result = response.json()
+    except getattr(json.decoder, 'JSONDecodeError', ValueError) as inst:
+        module.fail_json(msg="Invalid JSON response with error: %s" % inst)
+
+    if code not in success_codes:
+        msg = navigate_hash(result, ['message'])
+        if msg:
+            module.fail_json(msg=msg)
+        else:
+            module.fail_json(msg="operation failed, return code=%d" % code)
+
+    return result
+
+
+def is_different(expect, actual):
```

Comment:
This is something you might want to move into `module_utils/hwc_utils.py`

Reply:
Done in new patch https://github.com/ansible/ansible/pull/54102